File: history/0.1.2/tatala-client-csharp/src/QiLeYuan/Tatala/socket/to/helper/LongMapHelper.cs
using System;
using System.Collections.Generic;
using QiLeYuan.Tatala.socket.io;
using QiLeYuan.Tatala.socket.util;

namespace QiLeYuan.Tatala.socket.to.helper {

    public class LongMapHelper {
        public static int getByteArrayLength(Dictionary<String, Int64> longMap) {
            int blength = 0;
            foreach (KeyValuePair<String, Int64> item in longMap) {
                // type
                blength += TransferUtil.getLengthOfByte();
                // key
                String key = item.Key;
                blength += TransferUtil.getLengthOfString(key);
                // value
                blength += TransferUtil.getLengthOfLong();
            }
            return blength;
        }

        public static void getByteArray(Dictionary<String, Int64> longMap, TransferOutputStream touts) {
            foreach (KeyValuePair<String, Int64> item in longMap) {
                // type
                touts.writeByte(TransferObject.DATATYPE_LONG);
                // key
                String key = item.Key;
                touts.writeString(key);
                // value
                long value = item.Value;
                touts.writeLong(value);
            }
        }
    }
}

File: history/0.1.2/tatala-java/src/com/qileyuan/tatala/socket/to/helper/LongMapHelper.java
package com.qileyuan.tatala.socket.to.helper;

import java.util.Iterator;
import java.util.Map;

import com.qileyuan.tatala.socket.io.TransferOutputStream;
import com.qileyuan.tatala.socket.to.TransferObject;
import com.qileyuan.tatala.socket.util.TransferUtil;

public class LongMapHelper {
	public static int getByteArrayLength(Map<String, Long> longMap){
		int blength = 0;
		Iterator<String> keyIt = longMap.keySet().iterator();
		while (keyIt.hasNext()) {
			// type
			blength += TransferUtil.getLengthOfByte();
			// key
			String key = (String) keyIt.next();
			blength += TransferUtil.getLengthOfString(key);
			// value
			blength += TransferUtil.getLengthOfLong();
		}
		return blength;
	}
	
	public static void getByteArray(Map<String, Long> longMap, TransferOutputStream touts) {
		Iterator<String> keyIt = longMap.keySet().iterator();
		while (keyIt.hasNext()) {
			// type
			touts.writeByte(TransferObject.DATATYPE_LONG);
			// key
			String key = (String) keyIt.next();
			touts.writeString(key);
			// value
			long value = longMap.get(key);
			touts.writeLong(value);
		}
	}
}
